machine/v1beta1/types_vsphereprovider.go

```diff
@@ -87,12 +87,124 @@ type NetworkSpec struct {
 	Devices []NetworkDeviceSpec `json:"devices"`
 }
 
+type DHCPState string
+
+const (
+	EnabledState  DHCPState = "Enabled"
+	DisabledState DHCPState = "Disabled"
+)
+
+// AddressesFromPool is an IPAddressPool that will be used to create
+// IPAddressClaims for fulfillment by an external controller.
+type AddressesFromPool struct {
+	Group string `json:"group"`
+
+	// Kind is the type of resource being referenced
+	Resource string `json:"resource"`
+
+	// Name is the name of resource being referenced
+	Name string `json:"name"`
```

Comment:
Need to add validation tags to this, optional, required, any format that can be applied. Same goes for other fields in this struct as well please